crypto/x509v3/v3_lib.c

```diff
@@ -347,13 +347,22 @@ int X509V3_add1_i2d(STACK_OF(X509_EXTENSION) **x, int nid, void *value,
         return 1;
     }
 
-    if (!*x && !(*x = sk_X509_EXTENSION_new_null()))
-        return -1;
-    if (!sk_X509_EXTENSION_push(*x, ext))
-        return -1;
+    ret = *x;
+    if (!*x
```

Comment:
I think
```
if ((ret = *x) == NULL
 && ...
```
is better.  Okay if you disagree, but the ! must be replaced :)

Reply:
Fixed  !